### ISSUE TITLE
Incorrect error response when connecting merchant

### ISSUE THREAD

**Issue by mikkamp** [woocommerce/google-listings-and-ads#935]
### Describe the bug:
In our code when we are claiming the URL we check for the error 403 (and if it's a sub account) we send that error on to the UI including the URL details, indicating that the claim failed. [#code](https://github.com/woocommerce/google-listings-and-ads/blob/1.3.0/src/API/Site/Controllers/MerchantCenter/AccountController.php#L436-L450)

Our UI handles this and checks for a 403 and then shows the reclaim URL card stating that the claim failed. [#code](https://github.com/woocommerce/google-listings-and-ads/blob/1.3.0/js/src/setup-mc/setup-stepper/setup-accounts/google-mc-account/connect-mc-card/index.js#L74-L78)

However there are many other scenarios where connecting a merchant account could return a 403. One such case is when the user does not authorize the right scopes when connecting their Google account. This means it will fail before trying to claim the URL and the first recorded failure is when we try to get the site verification token with the error:

```
2021-07-26T15:45:08+00:00 ERROR Automattic\WooCommerce\GoogleListingsAndAds\API\Google\SiteVerification::get_token {
  "error": {
    "code": 403,
    "message": "Request had insufficient authentication scopes.",
    "errors": [
      {
        "message": "Insufficient Permission",
        "domain": "global",
        "reason": "insufficientPermissions"
      }
    ],
    "status": "PERMISSION_DENIED"
  }
}
```

Since that's also a 403 the UI allows the error to pass and responds with the same reclaim URL card, since the error doesn't contain any information about the website URL yet. We see the reclaim URL card with the message:

```
Your URL,, is currently claimed by another Merchant Center account.
```

Which is very confusing to the user because in most cases there is no other account which has claimed the URL.

Possible ways to handle this would be to confirm in the UI that it's a 403 and at the step `claim`, or alternatively we should use a different error code when we pass it to the UI.

### Expected behavior:
Show the correct reason why the connection failed. In this case sending through the error `Request had insufficient authentication scopes` would clearly indicate what's wrong.

### Steps to reproduce:
1. Go to the [security tab for your Google account](https://myaccount.google.com/security) and remove permissions for the WooCommerce app
2. Reconnect your Google account without authorizing the individual scopes (leave them unchecked)
3. Connect a merchant account
4. Observe the reclaim URL card with an empty website URL

### Additional details:
Also reported in [support thread](https://wordpress.org/support/topic/your-url-is-currently-claimed-by-another-merchant-center-account/)

**Comment by eason9487** [comment]
Dev note:
Probably this issue would be addressed after the partial oauth #1000 finished.

**Comment by findingsimple** [comment]
Good call @eason9487 - thanks!

**Comment by eason9487** [comment]
> Probably this issue would be addressed after the partial oauth #1000 finished.

Tested with adding partial oauth feature. There's a [permission scopes check](https://github.com/woocommerce/google-listings-and-ads/blob/77df8588f6b2e7a4ef4bf3a4d47cf92836845fce/js/src/setup-mc/setup-stepper/setup-accounts/index.js#L40-L42) before proceeding with merchant connection, so it will no longer encounter the **403** error - `Request had insufficient authentication scopes.`

The **403** error message prompting the user to reclaim their website is also displayed normally on UI, and the reclaiming works well.

![Kapture 2021-11-10 at 15 32 13](https://user-images.githubusercontent.com/17420811/141083958-e346e1e9-02f5-46cb-ba66-39e032336f13.gif)

I think it should be ok to close this issue.
